internal/clients/http/ec2/ec2_client.go

```diff
@@ -408,13 +408,11 @@ func (c *ec2Client) RunInstances(ctx context.Context, params *clients.AWSInstanc
 		},
 	}
 
-	if name != nil {
-		t := types.Tag{
-			Key:   ptr.To("Name"),
-			Value: name,
-		}
-		input.TagSpecifications[0].Tags = append(input.TagSpecifications[0].Tags, t)
+	t := types.Tag{
+		Key:   ptr.To("Name"),
+		Value: &name,
```

Comment:
Actually do we know what this does?
I'm guessing we don't want to have blank tag here, right?
So if it would end up tagging the instance with `name: ''`, I'd keep the condition, just checked for empty string instead of nil